exporters/elasticsearch/include/opentelemetry/exporters/elasticsearch/es_log_recordable.h

```diff
@@ -0,0 +1,177 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <map>
+#include <unordered_map>
+#include "nlohmann/json.hpp"
+#include "opentelemetry/sdk/common/attribute_utils.h"  // same as traces/attribute_utils
+#include "opentelemetry/sdk/logs/recordable.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+
+/**
+ * An Elasticsearch Recordable implemenation that stores the 10 fields of the Log Data Model inside
+ * a JSON object,
+ */
+class ElasticSearchRecordable final : public sdk::logs::Recordable
+{
+private:
+  // Define a JSON object that will be populated with the log data
+  nlohmann::json json_;
+
+  /**
+   * Converts a common::AttributeValue into a string, which is used for converting a
+   * common::AttributeValue into a string
+   */
+  const std::string ValueToString(const common::AttributeValue &value)
+  {
+    switch (value.index())
+    {
+      case common::AttributeType::TYPE_BOOL:
+        return (opentelemetry::nostd::get<bool>(value) ? "true" : "false");
+        break;
+      case common::AttributeType::TYPE_INT:
+        return std::to_string(opentelemetry::nostd::get<int>(value));
+        break;
+      case common::AttributeType::TYPE_INT64:
+        return std::to_string(opentelemetry::nostd::get<int64_t>(value));
+        break;
+      case common::AttributeType::TYPE_UINT:
+        return std::to_string(opentelemetry::nostd::get<unsigned int>(value));
+        break;
+      case common::AttributeType::TYPE_UINT64:
+        return std::to_string(opentelemetry::nostd::get<uint64_t>(value));
+        break;
+      case common::AttributeType::TYPE_DOUBLE:
+        return std::to_string(opentelemetry::nostd::get<double>(value));
+        break;
+      case common::AttributeType::TYPE_STRING:
+      case common::AttributeType::TYPE_CSTRING:
+        return opentelemetry::nostd::get<opentelemetry::nostd::string_view>(value).data();
+        break;
+      default:
+        return "Invalid type";
+        break;
+    }
+  }
+
+public:
+  /**
+   * Set the severity for this log.
+   * @param severity the severity of the event
+   */
+  void SetSeverity(opentelemetry::logs::Severity severity) noexcept override
+  {
+    // Convert the severity enum to a string
+    static opentelemetry::nostd::string_view severityStringMap_[25] = {
+        "kInvalid", "kTrace",  "kTrace2", "kTrace3", "kTrace4", "kDebug",  "kDebug2",
+        "kDebug3",  "kDebug4", "kInfo",   "kInfo2",  "kInfo3",  "kInfo4",  "kWarn",
+        "kWarn2",   "kWarn3",  "kWarn4",  "kError",  "kError2", "kError3", "kError4",
+        "kFatal",   "kFatal2", "kFatal3", "kFatal4"};
+    json_["severity"] = severityStringMap_[static_cast<int>(severity)];
```

Comment:
nit: I think `severity_num` and `severity_text` should both separated into 2 different fields and both added to the json object - following the spec's rec, having both is ideal :)

Reply:
hmmm it might be overkill for the user, since the severity probably isn't the main focus of the log statement. Having one field that says the severity is probably fine, but if anyone else feels strongly I can change it :)

Reply:
wouldn't elasticsearch need to filter/query by severity? maybe the severity number would be easier to compare than the text then?